src/agentscope/service/web/tripadvisor.py

```diff
@@ -0,0 +1,526 @@
+# -*- coding: utf-8 -*-
+"""TripAdvisor APIs for searching and retrieving location information."""
+
+from loguru import logger
+import requests
+from agentscope.service.service_response import ServiceResponse
+from agentscope.service.service_status import ServiceExecStatus
+
+
+def get_tripadvisor_location_photos(
+    api_key: str,
+    location_id: str = None,
+    query: str = None,
+    language: str = "en",
+) -> ServiceResponse:
+    """
+    Retrieve photos for a specific location using the TripAdvisor API.
+
+    Args:
+        api_key (`str`):
+            Your TripAdvisor API key.
+        location_id (`str`, optional):
+            The ID of the location for which to retrieve photos.
+            Required if query is not provided.
+        query (`str`, optional):
+            The search query to find a location. Required if
+            location_id is not provided.
+        language (`str`, optional):
+            The language for the response. Defaults to 'en'.
+
+    Returns:
+        `ServiceResponse`: A dictionary with two variables: `status` and
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is the JSON response from TripAdvisor API or error
+        information, which depends on the `status` variable.
+
+        If successful, the `content` will be a dictionary
+        with the following structure:
+        {
+            'photo_data': {
+                'data': [
+                    {
+                        'id': int,
+                        'is_blessed': bool,
+                        'caption': str,
+                        'published_date': str,
+                        'images': {
+                            'thumbnail': {
```

Comment:
What's the differences among these keys? e.g. thumbnail, small, media, large and original

Are they the same picture but only differ in size?

Reply:
Yes. According my testing that is indeed the case.